src/main/java/org/opensearch/knn/training/ByteTrainingDataConsumer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.training;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.knn.jni.JNICommons;
+import org.opensearch.knn.index.memory.NativeMemoryAllocation;
+import org.opensearch.search.SearchHit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Transfers byte vectors from JVM to native memory.
+ */
+public class ByteTrainingDataConsumer extends TrainingDataConsumer<byte[]> {
+
+    /**
+     * Constructor
+     *
+     * @param trainingDataAllocation NativeMemoryAllocation that contains information about native memory allocation.
+     */
+    public ByteTrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation) {
+        super(trainingDataAllocation);
+    }
+
+    @Override
+    public void accept(List<byte[]> byteVectors) {
```

Comment:
Why public?

Reply:
previously implemented from Consumer, which are public, now removed parent implement from Consumer and changed it to protected

Reply:
It can be even private. I don't see it being used outside of this class.

Reply:
accept method is used in some tests, but I removed other methods.

Reply:
Thanks for the update but could you try to remove accept method as well? It is not great to have it just because it is used in the test.

Reply:
The `accept` is also used in the processTrainingVectors.